public/app/features/alerting/unified/utils/time.ts

```diff
@@ -95,7 +95,68 @@ export function parsePrometheusDuration(duration: string): number {
   return totalDuration;
 }
 
-export const safeParseDurationstr = (duration: string): number => {
+export function formatPrometheusDuration(milliseconds: number): string {
+  const seconds = Math.floor(milliseconds / 1000);
+  const minutes = Math.floor(seconds / 60);
+  const hours = Math.floor(minutes / 60);
+  const days = Math.floor(hours / 24);
+  const weeks = Math.floor(days / 7);
+  const years = Math.floor(days / 365);
+
+  // we'll make an exception here for 0, 0ms seems a bit weird
+  if (milliseconds === 0) {
```

Comment:
May as well add a test for this case as well?

Reply:
Nice catch!